app/uk/gov/hmrc/thirdpartyapplication/controllers/ApplicationCommandController.scala

```diff
@@ -31,8 +29,24 @@ import uk.gov.hmrc.thirdpartyapplication.domain.models.{ApplicationCommand, Appl
 import uk.gov.hmrc.thirdpartyapplication.models.ApplicationResponse
 import uk.gov.hmrc.thirdpartyapplication.models.JsonFormatters._
 import uk.gov.hmrc.thirdpartyapplication.services._
+import uk.gov.hmrc.apiplatform.modules.common.domain.models.LaxEmailAddress
+import play.api.libs.json.Reads
+import uk.gov.hmrc.thirdpartyapplication.services.commands.CommandFailureJsonFormatters
+import uk.gov.hmrc.thirdpartyapplication.services.commands.CommandHandler
+import uk.gov.hmrc.thirdpartyapplication.services.commands.CommandFailures._
 
 object ApplicationCommandController {
+  case class DispatchRequest(command: ApplicationCommand, verifiedCollaboratorsToNotify: Set[LaxEmailAddress])
+
+  object DispatchRequest {
+    import ApplicationCommandFormatters._
+
+    val readsExactDispatchRequest: Reads[DispatchRequest] = Json.reads[DispatchRequest]
```

Comment:
What is an "exact" dispatch request or command? Does this need a comment or slightly different name? Does it mean it's only one or the other, so maybe "pure" is a better word?

Reply:
Pure has connotations that don't apply here IMO.  Let me see if I can reword there two readers.

Reply:
I like the new names.